backend/server/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java

```diff
@@ -127,7 +124,12 @@ private boolean isResourceExists(Session session, String id, LineageEntryType ty
       throws ModelDBException {
     switch (type) {
       case EXPERIMENT_RUN:
-        return experimentDAO.isExperimentRunExists(session, id);
+        var query =
+            session.createQuery(
+                "select count(*) From ExperimentRunEntity ere where ere.id = :experimentRunId AND ere.deleted = false");
```

Comment:
put the query in a try-with-resources structure

Reply:
it is a hibernate query so it will auto-close when the session will close. we can not add it with try-with-resources

Reply:
ah. hibernate. ok.